src/plugins/intel_gpu/src/runtime/memory_pool.cpp

```diff
@@ -309,25 +319,146 @@ void memory_pool::clear_pool_for_network(uint32_t network_id) {
 
 memory_pool::memory_pool(engine& engine) : _engine(&engine) { }
 
-void memory_pool::dump(uint32_t net_id) {
-    GPU_DEBUG_COUT << "Dump memory pool of network " << net_id << std::endl;
+inline float get_mb_size(size_t size) {
+    return (static_cast<float>(size) / (1024 * 1024));
+}
+
+inline float get_percentage(size_t size, size_t total_size) {
+    return (static_cast<float>(size) * 100.0f / total_size);
+}
+
+size_t memory_pool::get_total_mem_pool_size(allocation_type type) {
+    size_t total_mem_pool_size = 0;
+    for (auto mem : _non_padded_pool) {
+        if (mem.second._type == type)
+            total_mem_pool_size += mem.first;
+    }
+
+    for (auto mem : _padded_pool) {
+        for (auto record : mem.second) {
+            if (record._type == type)
+                total_mem_pool_size += record._memory->size();
+        }
+    }
+    for (auto mem : _no_reusable_pool) {
+        if (mem.second._type == type)
+            total_mem_pool_size += mem.first;
+    }
+    return total_mem_pool_size;
```

Comment:
How about to record the size when a new entry is allocated or released, instead of calculating every time?

Reply:
Done.